build/vega-lite-schema.json

```diff
@@ -560,34 +560,76 @@
+            {
+              "$ref": "#/definitions/ConditionalAxisProperty<(def-alias-1203634144-2468-2565-1203634144-0-2566|undefined|null)>"
+            }
+          ]
         },
         "gridDash": {
-          "description": "An array of alternating [stroke, space] lengths for dashed grid lines.",
-          "items": {
-            "type": "number"
-          },
-          "type": "array"
+          "anyOf": [
+            {
+              "description": "An array of alternating [stroke, space] lengths for dashed grid lines.",
+              "items": {
+                "type": "number"
+              },
+              "type": "array"
+            },
+            {
+              "$ref": "#/definitions/ConditionalAxisProperty<(number[]|undefined|null)>"
+            }
+          ]
         },
         "gridDashOffset": {
-          "description": "The pixel offset at which to start drawing with the grid dash array.",
-          "type": "number"
+          "anyOf": [
+            {
+              "description": "The pixel offset at which to start drawing with the grid dash array.",
+              "type": "number"
+            },
+            {
+              "$ref": "#/definitions/ConditionalAxisProperty<(number|undefined|null)>"
+            }
+          ]
         },
         "gridOpacity": {
-          "description": "The stroke opacity of grid (value between [0,1])\n\n__Default value:__ `1`",
-          "maximum": 1,
-          "minimum": 0,
-          "type": "number"
+          "anyOf": [
+            {
+              "description": "The stroke opacity of grid (value between [0,1])\n\n__Default value:__ `1`",
+              "maximum": 1,
+              "minimum": 0,
+              "type": "number"
+            },
+            {
+              "$ref": "#/definitions/ConditionalAxisProperty<(number|undefined|null)>"
+            }
+          ]
         },
         "gridWidth": {
-          "description": "The grid width, in pixels.\n\n__Default value:__ `1`",
-          "minimum": 0,
-          "type": "number"
+          "anyOf": [
+            {
+              "description": "The grid width, in pixels.\n\n__Default value:__ `1`",
+              "minimum": 0,
+              "type": "number"
+            },
+            {
+              "$ref": "#/definitions/ConditionalAxisProperty<(number|undefined|null)>"
+            }
+          ]
         },
         "labelAlign": {
-          "$ref": "#/definitions/Align",
-          "description": "Horizontal text alignment of axis tick labels, overriding the default setting for the current axis orientation."
+          "anyOf": [
+            {
+              "$ref": "#/definitions/Align",
+              "description": "Horizontal text alignment of axis tick labels, overriding the default setting for the current axis orientation."
+            },
+            {
+              "$ref": "#/definitions/ConditionalAxisProperty<(def-alias-981995915-5990-6039-981995915-0-9900|undefined|null)>"
```

Comment:
> def-alias-981995915-5990-6039-981995915-0-9900

@domoritz  I don't know why the JSON schema generator doesn't read the name `Align` from Vega and put it here. 

That said, the schema is still technically correct here.

Reply:
I'd need to look more into this but it's not an easy type: `[k in ConditionalAxisProp]?: BaseAxisNoSignals[k] | ConditionalAxisProperty<BaseAxisNoSignals[k]>;`.